src/jv_aux.c

```diff
@@ -38,29 +46,35 @@ static int parse_slice(jv j, jv slice, int* pstart, int* pend) {
       jv_get_kind(end_jv) != JV_KIND_NUMBER) {
     jv_free(start_jv);
     jv_free(end_jv);
-    return 0;
-  } else {
-    double dstart = jv_number_value(start_jv);
-    double dend = jv_number_value(end_jv);
+    return jv_invalid_with_msg(jv_string("Array/string slice indices must be integers"));
+  }
+  if (!jv_is_integer(start_jv) || !jv_is_integer(end_jv)) {
     jv_free(start_jv);
     jv_free(end_jv);
-    if (dstart < 0) dstart += len;
-    if (dend < 0) dend += len;
-    if (dstart < 0) dstart = 0;
-    if (dstart > len) dstart = len;
-
-    int start = (int)dstart;
-    int end = (dend > len) ? len : (int)dend;
-    // Ends are exclusive but e.g. 1 < 1.5 so :1.5 should be :2 not :1
-    if(end < dend) end += 1;
-
-    if (end > len) end = len;
-    if (end < start) end = start;
-    assert(0 <= start && start <= end && end <= len);
-    *pstart = start;
-    *pend = end;
-    return 1;
-  }
+    if (slice_for == FOR_READ)
+      return jv_null();
+    return jv_invalid_with_msg(jv_string("Array/string slice indices must be integers"));
+  }
+  double dstart = jv_number_value(start_jv);
+  double dend = jv_number_value(end_jv);
+  jv_free(start_jv);
+  jv_free(end_jv);
+  if (dstart < 0) dstart += len;
+  if (dend < 0) dend += len;
+  if (dstart < 0) dstart = 0;
+  if (dstart > len) dstart = len;
+
+  int start = (int)dstart;
```

Comment:
Like for the modulo operator, this is unspecified behaviour if `dstart` is an integer according  to `jv_is_integer()`, but it is bigger/smaller than `INT_MIN`/`INT_MAX`. I think we should add some `jv_int_value()` and `jv_intmax_value()` functions that round and clamp numbers consistently, and use those everywhere we need an integer instead of `(int)jv_number_value()`.

Reply:
Fixed.